src/flb_scheduler.c

```diff
@@ -350,6 +350,14 @@ int flb_sched_request_invalidate(struct flb_config *config, void *data)
         }
     }
 
```

Comment:
add documentation about the logic here, why does it need to clean up the request_wait